engine/src/main/java/org/terasology/engine/physics/bullet/BulletPhysics.java

```diff
@@ -330,133 +598,216 @@ public void update(float delta) {
         } catch (Exception e) {
             logger.error("Error running simulation step.", e);
         }
-        collisions.addAll(getNewCollisionPairs());
-    }
 
-    public btDiscreteDynamicsWorld getDiscreteDynamicsWorld() {
-        return this.discreteDynamicsWorld;
-    }
+        PerformanceMonitor.startActivity("Physics Update");
 
-    @Override
-    public boolean removeRigidBody(EntityRef entity) {
-        BulletRigidBody rigidBody = entityRigidBodies.remove(entity);
-        if (rigidBody != null) {
-            removeRigidBody(rigidBody);
-            // wake up this entities neighbors
+        // update collision pairs
```

Comment:
Please extract parts of the following code into separate methods where applicable... The whole `update()` method is way too big to be comprehensible.